dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResult.java

```diff
@@ -0,0 +1,374 @@
+    public boolean isCodeLink() {
+        return this.codeLink != null;
+    }
+
+
+    public String getCodeLink() {
+        return codeLink;
+    }
+
+    public void setCodeLink(String codeLink) {
+        this.codeLink = codeLink;
+    }
+
+
+
+    public int getOrder() {
+        return order;
+    }
+
+    public void setOrder(int order) {
+        this.order = order;
+    }
+
+
+
+
+    public void setChildren(List<NavResult> children) {
+        this.children = children;
+    }
+
+    public boolean isFolder() {
+        return folderId != null;
+    }
+
+    public List<? extends NavResult> getChildren() throws Exception {
+        if (children == null && hostId != null && folderId != null) {
+            // lazy loadinge children
+            Host host = APILocator.getHostAPI()
+                .find(hostId, sysuser, true);
+            Folder folder = APILocator.getFolderAPI()
+                .find(folderId, sysuser, true);
+            Identifier ident = APILocator.getIdentifierAPI()
+                .find(folder);
+            NavResult lazyMe = new NavTool().getNav(host, ident.getPath(), languageId, sysuser);
+
+            children = lazyMe.getChildren();
+            childrenFolderIds = lazyMe.getChildrenFolderIds();
+        }
+        if (children != null) {
+            ArrayList<NavResult> list = new ArrayList<NavResult>();
+            for (NavResult nn : children) {
+                if (nn.isFolder()) {
+                    // for folders we avoid returning the same instance
+                    // it could be changed elsewhere and we need it to
+                    // load its children lazily
+                    NavResult ff = new NavResult(folderId, nn.hostId, nn.folderId, nn.languageId);
+                    ff.setTitle(nn.getTitle());
+                    ff.setHref(nn.getHref());
+                    ff.setOrder(nn.getOrder());
+                    ff.setType(nn.getType());
+                    ff.setPermissionId(nn.getPermissionId());
+                    ff.setShowOnMenu(nn.isShowOnMenu());
+                    list.add(ff);
+                } else {
+                    list.add(nn);
+                }
+            }
+            return list;
+        } else {
+            return new ArrayList<NavResult>();
+        }
+    }
+
+    public String getParentPath() throws DotDataException, DotSecurityException {
+        if (parent == null)
+            return null; // no parent! I'm the root folder
+        if (parent.equals(FolderAPI.SYSTEM_FOLDER))
+            return "/";
+        Folder folder = APILocator.getFolderAPI()
+            .find(parent, sysuser, true);
+        Identifier ident = APILocator.getIdentifierAPI()
+            .find(folder);
+        return ident.getURI();
+    }
+
+    public NavResult getParent() throws DotDataException, DotSecurityException {
+        String path = getParentPath();
+        if (path != null) {
+            NavResult lazyMe = new NavTool().getNav(APILocator.getHostAPI()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "lazyMe". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)